core/src/main/java/org/openapitools/openapidiff/core/compare/schemadiffresult/ComposedSchemaDiffResult.java

```diff
@@ -88,10 +90,14 @@ public ComposedSchemaDiffResult(OpenApiDiff openApiDiff) {
     }
   }
 
-  private Map<String, Schema> getSchema(Components components, Map<String, String> mapping) {
+  private Map<String, Schema> getSchema(Components components, Map<String, String> mapping, ComposedSchema composedSchema) {
     Map<String, Schema> result = new LinkedHashMap<>();
     mapping.forEach(
         (key, value) -> result.put(key, refPointer.resolveRef(components, new Schema(), value)));
+
+    result.putAll(getUnnamedSchemas(composedSchema.getAllOf(), "all-of"));
+    result.putAll(getUnnamedSchemas(composedSchema.getOneOf(), "one-of"));
+    result.putAll(getUnnamedSchemas(composedSchema.getAnyOf(), "any-of"));
```

Comment:
quick and dirty workaround - naming the unnamed schemas in order of their appearance in the list.